mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx

```diff
@@ -39,49 +51,50 @@ describe('SchemaTable', () => {
   });
 
   test('should render with minimal props without exploding', () => {
-    wrapper = mountWithIntl(<SchemaTable {...minimalProps} />);
-    expect(wrapper.length).toBe(1);
+    const { container } = renderWithIntl(<SchemaTable {...minimalProps} />);
+    expect(container).not.toBeNull();
   });
 
   test('should nested table not be rendered by default', () => {
-    wrapper = mountWithIntl(
+    const { container } = renderWithIntl(
       <MemoryRouter>
         <SchemaTable {...props} />
       </MemoryRouter>,
     );
-    expect(wrapper.find(Table).length).toBe(1);
-    expect(wrapper.find('.outer-table').find(Table).length).toBe(1);
-    expect(wrapper.find('.inner-table').find(Table).length).toBe(0);
-    expect(wrapper.html()).toContain('Inputs');
-    expect(wrapper.html()).toContain('Outputs');
-    expect(wrapper.html()).toContain('Name');
-    expect(wrapper.html()).toContain('Type');
-    expect(wrapper.html()).not.toContain('column1');
-    expect(wrapper.html()).not.toContain('string');
-    expect(wrapper.html()).not.toContain('score1');
-    expect(wrapper.html()).not.toContain('long');
+    expect(container.querySelector('.outer-table table')).not.toBeNull();
+    expect(container.querySelector('.inner-table table')).toBeNull();
+    expect(container.innerHTML).toContain('Inputs');
+    expect(container.innerHTML).toContain('Outputs');
+    expect(container.innerHTML).toContain('Name');
+    expect(container.innerHTML).toContain('Type');
+    expect(container.innerHTML).not.toContain('column1');
+    expect(container.innerHTML).not.toContain('string');
+    expect(container.innerHTML).not.toContain('score1');
+    expect(container.innerHTML).not.toContain('long');
```

Comment:
Just a side note: recommended (more BDD) approach with RTL (contrary to Enzyme) is to use `screen` object imported from the same package as `render`, then build assertions like
```
expect(screen.getByRole('cell', { name: 'Inputs' })).toBeInTheDocument();
```
In general, we try to use accessible attributes (roles, names, labels) in RTL tests in order to make the test as high level (thus closer to real user interaction) as possible. However, this ☝️ is pretty valid here as well esp. given the fact we're just migrating 1:1 from `enzyme` :)

Reply:
I'll definitely keep this in mind for the future! It looks like in order to migrate to the `screen` API we'd probably have to add quite a few `data-testid`/`role`s to the code (which would be a good thing to do, but perhaps in a future PR)